torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -152,7 +153,12 @@ void UnrollPass::handle(kir::ForLoop* fl) {
   if (!non_trivial_pred_found_) {
     loop_replacement_map_.insert({fl, inlined_loop});
   } else {
-    unroll_ite->elseBody().push_back(inlined_loop);
+    kir::ExpressionEvaluator eval;
+    const auto result = eval.evaluate(fl->iter_domain()->rawExtent());
+    // No need to generate the else part if the extent is 1
+    if (result.has_value() && result.value() != 1) {
```

Comment:
Should this be `!(result.has_value() && result.value() == 1)`? Not sure it makes a difference in practice as we shouldn't unroll a loop based on a non constant value, but seems generally safer.

Reply:
Oops, thanks. Fixed.